fink_utils/sso/spins.py

```diff
@@ -1092,7 +1104,10 @@ def fit_spin(
         return outdic
 
     # For the chi2, we use the error estimate from the data directly
-    chisq = np.sum((res_lsq.fun / sigmapsf) ** 2)
+    _, sorted_sigmapsf = zip(*sorted(zip(filters, sigmapsf)))  # Sorted by filter
+    sorted_sigmapsf = np.array(sorted_sigmapsf)
```

Comment:
Nice trick. As you are doing the same operations as above, can you factorize the code by defining a function that does the sorting? e.g. in the lines of:

```python
def some_name(indices, vector):
  _, sorted_vector = zip(*sorted(zip(indices, vector)))
  return np.array(sorted_vector)
```